pilot/pkg/config/monitor/monitor.go

```diff
@@ -52,29 +53,60 @@ func NewMonitor(name string, delegateStore model.ConfigStore, getSnapshotFunc fu
 	return monitor
 }
 
+const watchDebounceDelay = 50 * time.Millisecond
+
+// Trigger notifications when a file is mutated
+func fileTrigger(path string, ch chan struct{}, stop <-chan struct{}) error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return err
+	}
+	if err = watcher.Add(path); err != nil {
+		return err
+	}
+	go func() {
+		defer watcher.Close()
+		var debounceC <-chan time.Time
+		for {
+			select {
+			case <-debounceC:
+				debounceC = nil
+				ch <- struct{}{}
+			case <-watcher.Events:
+				if debounceC == nil {
+					debounceC = time.After(watchDebounceDelay)
+				}
+			case err := <-watcher.Errors:
+				log.Warnf("Error watching file trigger: %v %v", path, err)
+				return
```

Comment:
should you return this error?

Reply:
This is running async so not sure where we would return the error to?